sdks/go/pkg/beam/options/jobopts/options_test.go

```diff
@@ -0,0 +1,122 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jobopts
+
+import (
+	"context"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core"
+)
+
+func TestGetEndpoint(t *testing.T) {
+	address := "localhost:8099"
+	Endpoint = &address
+	v, err := GetEndpoint()
+	if err != nil {
+		t.Fatalf("unexpected error GetEndpoint(): %v", err)
+	}
+	if v != address {
+		t.Errorf("GetEndpoint() = %v, want %v", v, address)
+	}
+}
+
+func TestGetEndpoint_Bad(t *testing.T) {
+	address := ""
+	Endpoint = &address
+	expectedErr := "no job service endpoint specified. Use --endpoint=<endpoint>"
```

Comment:
If you're going to test the error, I would strongly recommend against string comparison. I recommend creating a custom error type and returning that, and then in the test do type assertion to see if it matches that type. (Example here: https://quii.gitbook.io/learn-go-with-tests/questions-and-answers/error-types)

In this case, it doesn't even have to be a custom struct. You can just do something like `type missingFlagError error`.  Then return the error as that new type and type assert here.

(As an aside, I think it's strange that this one function specifically returns an error while all the others just return the empty values, but it's too late to change that now.)

Reply:
Done